### PR TITLE
Spark2.0 scripts added

### PR DESCRIPTION
**Pull request by Georgehe4**

### CODE REVIEW

File: scripts/jenkins-test [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+echo "Testing MANGO version ${VERSION} on Spark ${SPARK_VERSION} and Hadoop ${HADOOP_VERSION}"
+
+# first, build the sources, run the unit tests, and generate a coverage report
+mvn clean \
+    -Dhadoop.version=${HADOOP_VERSION} \
+    -Dspark.version=${SPARK_VERSION}
+
+# if those pass, build the distribution package and the integration tests
+mvn test \
+    -P coverage scoverage:report
```
Comment (akmorrow13) [inline scripts/jenkins-test]: remove this

**Comment by akmorrow13** [comment]
Jenkins, test this please.

**Comment by Georgehe4** [comment]
Jenkins, retest this please

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala [hunk 2]
```diff
@@ -562,28 +562,28 @@ class VizReads(protected val args: VizReadsArgs) extends BDGSparkCommand[VizRead
...
+            sc.parallelize[(ReferenceRegion, Long)](Array[(ReferenceRegion, Long)]())
           } else {
-            var variants: RDD[Genotype] = VizReads.sc.emptyRDD[Genotype]
+            var variants: RDD[Genotype] = VizReads.sc.parallelize[(Genotype)](Array[(Genotype)]())
             VizReads.variantData.get.files.foreach(fp => variants = variants.union(GenotypeMaterialization.load(sc, None, fp)))
             val threshold = args.threshold
             GenotypeFilter.filter(variants, GenotypeFilterType(variantFilter.get), VizReads.chunkSize, threshold)
           }
-        } else sc.emptyRDD[(ReferenceRegion, Long)]
+        } else sc.parallelize[(ReferenceRegion, Long)](Array[(ReferenceRegion, Long)]())
```
Comment (Georgehe4) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: Addressed.

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala [hunk 3]
```diff
@@ -562,28 +562,28 @@ class VizReads(protected val args: VizReadsArgs) extends BDGSparkCommand[VizRead
         if (variantFilter.isDefined) {
           if (!VizReads.variantsExist) {
             log.warn("specified discovery predicate for variants but no variant files were provided")
-            sc.emptyRDD[(ReferenceRegion, Long)]
+            sc.parallelize[(ReferenceRegion, Long)](Array[(ReferenceRegion, Long)]())
```
Comment (Georgehe4) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: Addressed.

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala [hunk 4]
```diff
@@ -562,28 +562,28 @@ class VizReads(protected val args: VizReadsArgs) extends BDGSparkCommand[VizRead
...
           } else {
-            var features: RDD[Feature] = sc.emptyRDD[Feature]
+            var features: RDD[Feature] = sc.parallelize[(Feature)](Array[(Feature)]())
             VizReads.featureData.get.files.foreach(fp => features = features.union(FeatureMaterialization.load(sc, None, fp).rdd))
             val threshold = args.threshold
             FeatureFilter.filter(features, FeatureFilterType(featureFilter.get), VizReads.chunkSize, threshold)
           }
-        } else sc.emptyRDD[(ReferenceRegion, Long)]
+        } else sc.parallelize[(ReferenceRegion, Long)](Array[(ReferenceRegion, Long)]())
```
Comment (Georgehe4) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: Addressed.

File: mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala [hunk 5]
```diff
@@ -562,28 +562,28 @@ class VizReads(protected val args: VizReadsArgs) extends BDGSparkCommand[VizRead
...
 
       // filtering for features
       val featureRegions: RDD[(ReferenceRegion, Long)] =
         if (featureFilter.isDefined) {
           if (!VizReads.featuresExist) {
             log.warn("specified discovery predicate for features but no variant files were provided")
-            sc.emptyRDD[(ReferenceRegion, Long)]
+            sc.parallelize[(ReferenceRegion, Long)](Array[(ReferenceRegion, Long)]())
```
Comment (Georgehe4) [inline mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala]: Also addressed.

File: mango-core/src/main/scala/org/bdgenomics/mango/models/LazyMaterialization.scala [hunk 6]
```diff
@@ -138,14 +139,15 @@ abstract class LazyMaterialization[T: ClassTag](name: String) extends Serializab
...
 
+      var list = ListBuffer[RDD[(String, T)]]()
       // get alignment data for all samples
       files.map(fp => {
         val k = LazyMaterialization.filterKeyFromFile(fp)
         val d = load(region, fp).map(v => (k, v))
-        data = data.union(d)
+        list += d
       })
+      val data: RDD[(String, T)] = sc.union[(String, T)](list)
```
Comment (Georgehe4) [inline mango-core/src/main/scala/org/bdgenomics/mango/models/LazyMaterialization.scala]: Modified.

File: scripts/move_to_spark_1.6.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set +x
+
+find . -name "pom.xml" -exec sed \
+     -e "/scala\.version/ s/2\.11\.7/2.10.4/g" \
+     -e "/scala\.version\.prefix/ s/2\.11/2.10/g" \
+     -e "/spark\.version/ s/2\.0\.0/1.6.1/g" \
+     -e "/spark\.version\.prefix/ s/-spark2\_/_/g" \
+     -i.spark2.bak '{}' \;
```
Comment (Georgehe4) [inline scripts/move_to_spark_1.6.sh]: Added line

File: scripts/jenkins-test [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    -Dspark.version=${SPARK_VERSION} \
+    -DargLine=${MANGO_MVN_TMP_DIR}
+
+# we are done with maven, so clean up the maven temp dir
+find ${MANGO_MVN_TMP_DIR}
+rm -rf ${MANGO_MVN_TMP_DIR}
+
+echo
+echo "All the tests passed"
+echo
```
Comment (Georgehe4) [inline scripts/jenkins-test]: Added

File: scripts/move_to_spark_2.sh [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set +x
+
+find . -name "pom.xml" -exec sed \
+     -e "/scala\.version/ s/2\.10\.4/2.11.7/g" \
+     -e "/scala\.version\.prefix/ s/2\.10/2.11/g" \
+     -e "/spark\.version/ s/1\.6\.1/2.0.0/g" \
+     -e "/spark\.version\.prefix/ s/\_/-spark2_/g" \
+     -i.spark2.bak '{}' \;
```
Comment (Georgehe4) [inline scripts/move_to_spark_2.sh]: Added

**Comment by akmorrow13** [comment]
Thanks @Georgehe4 !

**Pull request merged**